exercises/pascals-triangle/example.rs

```diff
@@ -0,0 +1,25 @@
+pub struct PascalsTriangle {
+    row_count: u32,
+}
+
+impl PascalsTriangle {
+    pub fn new(row_count: u32) -> Self {
+        PascalsTriangle { row_count: row_count }
+    }
+
+    pub fn rows(&self) -> Vec<Vec<u32>> {
+        (0..self.row_count).map(|row| PascalsTriangle::row(row)).collect()
+    }
+
+    pub fn row(number: u32) -> Vec<u32> {
+        let mut r = vec![1];
+
+        for p in 1..(number + 1) {
+            if let Some(last) = r.pop() {
+                let next = last as f32 * ((number as f32 + 1f32 - p as f32) / p as f32);
```

Comment:
I am missing some math concept here. 

I had to introduce the floats so that it would properly determine the 3rd element in row 2. With ints, the formula looked like

```rust
let next = 2 * ((2 + 1 - 2) / 2);
```

or

```rust
let next = 2 * (1 / 2);
```

With ints, `1/2` returns 0. So my 2nd row became `[1,2,0]`

What am I missing here?